server/src/main/java/org/elasticsearch/plugins/DiscoveryPlugin.java

```diff
@@ -76,13 +76,15 @@ default Map<String, Supplier<SeedHostsProvider>> getSeedHostProviders(
      * join attempt but might be called multiple times during the lifetime of a node. Validators are expected to throw a
      * {@link IllegalStateException} if the node and the cluster-state are incompatible.
      */
+    @Deprecated
     default BiConsumer<DiscoveryNode, ClusterState> getJoinValidator() {
         return null;
     }
 
     /**
      * Allows plugging in election strategies (see {@link ElectionStrategy}) that define a customized notion of an election quorum.
      */
+    @Deprecated
```

Comment:
These deprecations will be added as part of https://github.com/elastic/elasticsearch/pull/88925